### ISSUE TITLE
Some pages becomes randomly white useing next.js with ISR.

### ISSUE THREAD

**Issue by emilpriverrc** [aws-amplify/amplify-hosting#2744]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d1bxok73m14yc9

### Region

eu-north-1

### Describe the bug

Randomly did we get white pages on the website which is using next.js. I have been debugging and do see that those pages have 0 bytes in the s3 bucket. Look at the photo. This happens randomly and 1 way to solve it is by removing the HTML file and it will be rebuilt and published.

I think that the invalidation is ok and that next.js revalidates the page successfully but when AWS pushes the file into s3 bucket it fail and I don't see any error logs on this and so on.

![Skärmavbild 2022-05-10 kl  14 24 59](https://user-images.githubusercontent.com/48241986/167627655-61726942-68fe-45ac-9bdc-d4e436e3c528.png)

### Expected behavior

Page to revalidated successfully and pushed into s3 bucket without that the file will be blank and have 0 bytes.

### Reproduction steps

It's hard to reproduce. The only way I see is to trigger a lot of revalidations and check for blank pages.

### Build Settings

```yaml
next.js default
```

**Comment by hloriana** [comment]
Hi @emilpriverrc 👋🏽 thanks for raising this issue. Can you provide a code snippet for how you're implementing ISR and also a screenshot of your next.config.js file while we investigate?

**Comment by emilpriverrc** [comment]
Sure!

next.config.js
```javascript
const withBundleAnalyzer = require("@next/bundle-analyzer")({
  enabled: process.env.ANALYZE === "true",
});

module.exports = withBundleAnalyzer({
  future: {
    webpack5: true,
  },
  i18n: {
    locales: ["en", "no", "de", "sv"],
    defaultLocale: "en",
  },
  images: {
    domains: [
        "a list of domains. i hide for the screenshot"
    ],
    deviceSizes: [320, 640, 750, 828, 1080, 1200, 1920, 2048, 2560, 3840],
    imageSizes: [16, 32, 48, 64, 96, 128, 256],
  },
  async rewrites() {
    return [
      {
        source: '/sitemap-en.xml',
        destination: '/api/sitemap-en',
      },
      {
        source: '/sitemap-no.xml',
        destination: '/api/sitemap-no',
      },
      {
        source: '/sitemap-de.xml',
        destination: '/api/sitemap-de',
      },
    ]
  },
  async headers() {
    return [
      {
        source: '/.well-known/apple-developer-merchantid-domain-association',
        headers: [
          {
            key: 'Content-Type',
            value: 'text/plain',
          },
        ],
      },
    ];
  },
});
```

Example of getStaticProps: 
```javascript
export const getStaticProps: GetStaticProps = async ({ params, preview = false, locale }) => {
  const { slug } = params as { slug: string };
  const language = (locale)?.toUpperCase() ?? 'EN';

  const primaryMenu = await getMenu(
    locale === 'en' ? 'PRIMARY' : `PRIMARY___${language}`,
  );
  const secondaryMenu = await getMenu(
    locale === 'en' ? 'SECONDARY' : `SECONDARY___${language}`,
  );
  const saleMenu = await getMenu(
    locale === 'en' ? 'SALE' : `SALE___${language}`,
  );
  const footerMenu1 = await getMenu(
    locale === 'en' ? 'FOOTER1' : `FOOTER1___${language}`,
  );
  const footerMenu2 = await getMenu(
    locale === 'en' ? 'FOOTER2' : `FOOTER2___${language}`,
  );
  const socialMenu = await getMenu(
    locale === 'en' ? 'SOCIAL' : `SOCIAL___${language}`,
  );

  const menu = {
    primary: primaryMenu,
    secondary: secondaryMenu,
    sale: saleMenu,
    footer1: footerMenu1,
    footer2: footerMenu2,
    social: socialMenu,
  };

  let [page] = await getPage(slug, language);
  const posts = await getAllPosts(language);
  const tags = await getAllTags(language);
  const themeSettings = await getThemeSettings();

  if (preview) {
    page = await getPreviewPost(slug) as Page;
  }

  if (!page) {
    return {
      notFound: true,
      revalidate: 60,
    };
  }

  return {
    props: {
      menu,
      page,
      posts,
      tags,
      themeSettings,
      translations: page?.translations?.map((translation) => ({
        uri: translation?.slug ?? null,
        locale: translation?.language?.code ?? null,
      })) ?? [],
    },
    revalidate: 60,
  };
};
```

Example of getStaticPaths
```javascript
export const getStaticPaths: GetStaticPaths = async () => {
  const pages = await getAllPages();

  const paths = pages.map((page) => ({
    params: {
      slug: page.slug,
    },
    locale: page.language.slug,
  }));

  return {
    paths,
    fallback: 'blocking',
  };
};
```


Something good to add to the issue is that we use next version `10.1.2`

**Comment by hloriana** [comment]
@emilpriverrc thank you for providing these details, we're taking a look.

**Comment by emilpriverrc** [comment]
Hello. Just wondering. Any update on this?

**Comment by emilpriverrc** [comment]
Hi again ✌️ we just got another site with exact same problem. This site is a copy on another page with same code and content just new amplify app. 

Should I send you some id for amplify ?

**Comment by hloriana** [comment]
Hi, we were also not able to reproduce this. Can you provide the logs for the ISR lambda function please? You can follow this FAQ guide to locate the logs: [Access Lambda Edge Logs](https://github.com/aws-amplify/amplify-hosting/blob/main/FAQ.md#access-lambda-edge-logs)

**Comment by hloriana** [comment]
Yes if you can provide that amplify app ID as well that would be helpful. When the app is first built, are the pages accessible? And is it only after the revalidation period that the files are 0kb?

**Comment by emilpriverrc** [comment]
Correct, It works fine after a build. It's first on revalidation this happens.

APP id: `d335y2aqqe2ekl`

Is it safe to post the logs here? Or can i send it to you in some other way?

**Comment by emilpriverrc** [comment]
I went through all the logs in cloud watch and the only thing I saw was that we loaded .env.production. But 0 errors and so on

**Comment by emilpriverrc** [comment]
Hi, we removed revalidation in getStaticProps and have not had any issues since then. Really looks like revalidation it's the problem

**Comment by ovdahlberg** [comment]
We have the same issue on multiple sites on different AWS accounts.

After build their is 0 byte pages and works after re-build.

Any update on this?

**Comment by CauanCabral** [comment]
Hi! 

We also have the same issue.

How we can get that behavior:

 1. Create a Next.js page with revalidation every X minutes
 2. At `getStaticProps` of your page, do some call to an external resource that can be unavailable (high traffic, network failure, random crashes) and throw an error.
 3. Next.js fail to generate that page.
 4. For an unknown reason, Amplify assume that the result is a success-generated page and delivers it to S3/Cloudfront.

After months of investigations, we continue to don't understand why AWS Amplify works that way.

We only could "workaround" that issue by making the page return some content (text/string) instead of just `throwing` the error to Next.js. But it is still useless.

Seems to us that Amplify makes 2 different tries to generate that page: one using the correct SSR and trying to generate a full static page. Other rendering the page/components with default values and relying on Next.js to do the hydration at the client-side with results of `getStaticProps`.

- Is that correct?
- Why Amplify doesn't understand a failed revalidation and just discard the result? Keeping to clients the old and correct content.

**Comment by rhoiyds** [comment]
I've also been experiencing this for months and haven't been able to find a suitable workaround to prevent 'Application Error' pages been shown on production when trying to load these 0kb pages.
At this point I'm considering automating a script to periodically scan the S3 bucket for 0KB static pages and deleting them - however this feels like a horrible band-aid.

**Comment by AnirudhaGohokar** [comment]
Facing same issue with nextjs using SSR. blank page coming sometimes specially when doing fast redirections.

**Comment by AnirudhaGohokar** [comment]
I was able resolve random blank page issue with NextJS 13.4.0. Getting error with latest version. So right now sticking with this

**Comment by pranavaccion** [comment]
@AnirudhaGohokar  still error exists there I  refresh those pages 500
Internal Server Error.

**Comment by andre-mr** [comment]
> Hi!
> 
> We also have the same issue.
> 
> How we can get that behavior:
> 
> 1. Create a Next.js page with revalidation every X minutes
> 2. At `getStaticProps` of your page, do some call to an external resource that can be unavailable (high traffic, network failure, random crashes) and throw an error.
> 3. Next.js fail to generate that page.
> 4. For an unknown reason, Amplify assume that the result is a success-generated page and delivers it to S3/Cloudfront.
> 
> After months of investigations, we continue to don't understand why AWS Amplify works that way.
> 
> We only could "workaround" that issue by making the page return some content (text/string) instead of just `throwing` the error to Next.js. But it is still useless.
> 
> Seems to us that Amplify makes 2 different tries to generate that page: one using the correct SSR and trying to generate a full static page. Other rendering the page/components with default values and relying on Next.js to do the hydration at the client-side with results of `getStaticProps`.
> 
> * Is that correct?
> * Why Amplify doesn't understand a failed revalidation and just discard the result? Keeping to clients the old and correct content.

Hi.
After this error occurs, does page display content again for new requests in the next revalidation cycle, if successful, without manual intervention?

**Comment by CauanCabral** [comment]
Hi @andre-mr ,

Yeah, after some revalidation cycles, the page returns without any intervention.

But we never know if it will take 30 minutes, 1 hour, 2 hours or more.

**Comment by FreshImmuc** [comment]
I just got a similar issue, page just white at some point. It somehow got fixed by restarting the dev server. I did not check how much bytes etc. were transferred.
